core/src/main/scala/io/github/ghostbuster91/sttp/client3/json/circe/CirceCoproductCodecGenerator.scala

```diff
@@ -116,11 +119,13 @@ private[circe] class CirceCoproductCodecGenerator() {
 
   private def decoderCasesWithMapping(
       discriminator: Discriminator[_],
-      failureTpe: Type.Name
+      failureTpe: Type.Name,
+      implicitMapping: Map[String, ClassName]
   ): List[Case] = {
     val mappedCases = discriminator match {
       case Discriminator.StringDsc(_, mapping) =>
-        mapping.toList.sortBy(_._1).map { case (k, v) =>
+        val mergedMapping = mergeMappings(mapping, implicitMapping)
+        mergedMapping.toList.sortBy(_._1).map { case (k, v) =>
```

Comment:
> For the enum type the mapping has to be complete, because we cannot implicitly determine the correct enum value for a child class.

:+1: 

> For the int type I am unsure how to approach it. And I am wondering id an int type property can correctly fit within the OpenApi spec.

I am not sure about that either. 
Openapi docs says:
> Mapping keys MUST be string values, but tooling MAY convert response values to strings for comparison.

I think that we can leave it as it is since using an `int` as a discriminator is quite rare case anyway. We can always return to that in the future if there will be a need, or once we will get more information.